pkg/sql/catalog/descs/temporary_descriptors.go

```diff
@@ -24,19 +24,57 @@ import (
 )
 
 type temporaryDescriptors struct {
-	codec       keys.SQLCodec
-	sessionData *sessiondata.SessionData
+	codec                   keys.SQLCodec
+	temporarySchemaProvider TemporarySchemaProvider
```

Comment:
nit: call call this field something shorter like provider, we're already in the context of temporary stuff.